### PR TITLE
Attributes Sourcemap & AST changes

### PR DESCRIPTION
**Pull request by pksunkara**
1. Added source maps for `Attributes`
2. Changed the structure for `DataStructure` and thus `Attributes` to be up to date with latest spec.
3. Serialised `Resource`, `Payload` & `Action` in accordance with the new spec.

Debt: We haven't changed the actual structures of `Resource`, `Payload` and `Action` to represent the exact structures in the latest spec.

For @zdne to review & merge.

### CODE REVIEW

**Comment by pksunkara** [comment]
@zdne This is ready to go.

File: features/fixtures/ast.json [hunk 1]
```diff
@@ -15,18 +15,20 @@
...
       "resources": [
         {
+          "element": "resource",
           "name": "<resource name>",
           "description": "<resource description>\n\n",
           "uriTemplate": "/resource/{parameter}",
           "model": {
             "name": "<resource name>",
-            "description": "<resource model description>",
+            "description": "<resource model description>\n",
```
Comment (zdne) [inline features/fixtures/ast.json]: Why this change?

Comment (pksunkara) [inline features/fixtures/ast.json]: Just making a small [fix](https://github.com/apiaryio/snowcrash/pull/281/files#diff-d3c5f2fc18729a968dbcb8059325dab9R17) in the blueprint file led to this.

File: src/BlueprintSourcemap.h [hunk 2]
```diff
@@ -101,7 +101,7 @@ namespace snowcrash {
         /** Source Map of Payload-specific Headers */
         SourceMap<Headers> headers;
 
-        /** Source Map of Payload-specific Attributes */
+        /** Source Map of Payload-specific Attributes (THIS SHOULD NOT BE HERE - should be under content) */
```
Comment (zdne) [inline src/BlueprintSourcemap.h]: we are going to pay for this.. one day ... :see_no_evil:

Comment (pksunkara) [inline src/BlueprintSourcemap.h]: Trust me, This is better for now.

File: features/fixtures/ast.json (hunk 1 above)
Comment (zdne) [inline features/fixtures/ast.json]: You mean the extra new line in the source blueprint led to this?

Comment (pksunkara) [inline features/fixtures/ast.json]: It was 

```
+ Model
    <resource model description>
```

The description would be considered a part of the `Model` markdown node. I just added a new line in between so that description is a new paragraph markdown node.

**Comment by zdne** [comment]
@pksunkara well, I am merging this but ... I am scared, ha ha
